### DISCUSSION TITLE
x11-selection and x11-cut-buffers

### DISCUSSION THREAD

**Discussion by skysky97** [kovidgoyal/kitty#6130] [Q&A] [answered]
As discussed in this: https://github.com/kovidgoyal/kitty/discussions/4893, contents can't be pasted after vim exit.

But vim said:

```
Whenever Vim is about to become unavailable (either via exiting or becoming
suspended), and thus unable to respond to another application's selection
request, it writes the contents of any owned selection to CUT_BUFFER0.
```

Does it means kitty do not use x11-cut-buffers?

**Comment by kovidgoyal** [comment, marked as answer]
Install the vim osc-yank plugin. Or better switch to neovim which
support proper OSC52 based copy/paste out of the box.

**Reply by skysky97** [reply]
> Install the vim osc-yank plugin. Or better switch to neovim which support proper OSC52 based copy/paste out of the box.

Thanks, I'll give it a try. And here I just have some interest in how the copy/paste works. After some search I got the iterms "x11-selections" "cut-buffers", and now "OSC52".

**Reply by kovidgoyal** [reply]
see https://sw.kovidgoyal.net/kitty/kittens/clipboard/
